src/Microsoft.Windows.CsWin32/Generator.cs

```diff
@@ -2874,6 +2874,15 @@ private StructDeclarationSyntax DeclareTypeDefStruct(TypeDefinition typeDef)
                 .WithExpressionBody(ArrowExpressionClause(AssignmentExpression(SyntaxKind.SimpleAssignmentExpression, fieldAccessExpression, valueParameter)))
                 .WithSemicolonToken(Token(SyntaxKind.SemicolonToken)));
 
+            if (fieldInfo.FieldType is IdentifierNameSyntax { Identifier: { Value: "nint" or "nuint" or nameof(IntPtr) or nameof(UIntPtr) } })
```

Comment:
I expected this to be `is PrimitiveTypeSyntax { IsNint: true } or { IsNuint: true }` but I saw that PrimitiveTypeHandleInfo is constructing IdentifierName instead of PredefinedType. I couldn't see a way to create such a PrimitiveTypeSyntax though.

Reply:
Never mind, just saw the definition of IsNint. 😆
http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/InternalSyntax/TypeSyntax.cs